src/main.zig

```diff
@@ -52,8 +52,8 @@ pub fn serialize(comptime T: type, allocator: Allocator, data: T, list: *ArrayLi
                         comptime var i = 0;
                         comptime var length = @sizeOf(T);
                         inline while (i < length_length) : (i += 1) {
-                            try list.append(@as(u8, @truncate(length)));
-                            length >>= 8;
+                            const shift = 8 * (length_length - 1 - i);
+                            try list.append(@as(u8, @truncate(length >> shift)));
```

Comment:
We should write the length in big endian format.

Reply:
fine but could we please have is as:

```
                         try list.append(183 + length_length);
-                        comptime var i = 0;
+                        comptime var i = length_length - 1;
                         comptime var length = @sizeOf(T);
-                        inline while (i < length_length) : (i += 1) {
+                        inline while (i >= 0) : (i -= 1) {
                             try list.append(@as(u8, @truncate(length)));
                             length >>= 8;
                         }
```

Reply:
yeah that was a mistake and now I understand what went wrong, here's what I meant to do:

```
                         try list.append(183 + length_length);
-                        comptime var i = 0;
+                        const offset_past_header = list.items.len;
+                        comptime var i = length_length - 1;
+                        try list.appendNTimes(0, length_length);
                         comptime var length = @sizeOf(T);
-                        inline while (i < length_length) : (i += 1) {
-                            try list.append(@as(u8, @truncate(length)));
+                        inline while (i >= 0) : (i -= 1) {
+                            list.items[offset_past_header + i] = @as(u8, @truncate(length));
                             length >>= 8;
                         }
                     },
```

which is actually more complicated but only does one allocation.